kf_lib_data_ingest/target_apis/kids_first.py

```diff
@@ -269,3 +287,143 @@
         CONCEPT.SEQUENCING
     }
 }
+
+
+def _get_kidsfirst_schema(url, cached_schema_filepath=CACHED_SCHEMA_FILE):
+    """
+    Get schemas for entities in the target API.
+
+    This defines the types for various properties of target concepts. The
+    property types will help us decide how to handle null values later on.
+
+    :param url: URL to an instance of the Kids First Data Service
+    :param cached_schema_filepath: file path to a JSON file containing a
+    saved version of the Kids First Data Service schema.
+    :returns output: a dict combining content from Data Service /status
+    endpoint and /swagger endpoint
+    """
+    # Read from file if it exists
+    if os.path.isfile(cached_schema_filepath):
+        return read_json(cached_schema_filepath)
+
+    output = None
+    err = None
+    common_msg = 'Unable to retrieve target schema from target service!'
+    status_url = f'{url}/status'
+    schema_url = f'{url}/swagger'
+
+    # Try to get schemas and version from kf data service
+    try:
+
+        status_response = requests_retry_session().get(status_url)
+        schema_response = requests_retry_session().get(schema_url)
+
+    except ConnectionError as e:
+
+        err = (f'{common_msg}\nCaused by {str(e)}')
+
+    else:
+        if ((status_response.status_code == 200) and
+                (schema_response.status_code == 200)):
+            # API version
+            version = {
+                'version': status_response.json().get('version'),
+                'commit': status_response.json().get('commit')
+            }
+            # Schemas
+            defs = schema_response.json()['definitions']
+            schemas = {
+                k: defs[to_camel_case(k)]
+                for k in target_concepts
+                if to_camel_case(k) in defs
+            }
+            # Add version to output
+            output = {
+                'version': version,
+                'definitions': schemas
+            }
+            # Update cache file
+            write_json(output, cached_schema_filepath)
+        else:
+            err = f'{common_msg}\nCaused by unexpected response(s):'
+            if status_response.status_code != 200:
+                err += f'\nFrom {status_url}:\n{status_response.text}'
+            if schema_response.status_code != 200:
+                err += f'\nFrom {schema_url}/swagger:\n{schema_response.text}'
+
+    # Report error
+    if err:
+        if logger:
+            logger.warning(err)
+        else:
+            print(err)
+
+    return output
+
+
+def handle_nulls(target_instances,
```

Comment:
This function doesn't seem like it only applies to the kids first dataservice.

Reply:
Yea I wasn't sure. As long as whatever is in `api_schema` dict still has the same structure. It should have top level `version` key and a `definitions` key. Since`definitions` has content from a swagger endpoint, the content should be standard for any swagger endpoint..i think